presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergMetadata.java

```diff
@@ -662,4 +702,335 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
         org.apache.iceberg.Table icebergTable = getIcebergTable(metastore, hdfsEnvironment, session, handle.getSchemaTableName());
         return TableStatisticsMaker.getTableStatistics(typeManager, session, constraint, handle, icebergTable);
     }
+
+    @Override
+    public void createMaterializedView(ConnectorSession session, SchemaTableName viewName, ConnectorMaterializedViewDefinition definition, boolean replace, boolean ignoreExisting)
+    {
+        Map<String, String> properties;
+
+        Map buildProps = new HashMap();
+        buildProps.put(PRESTO_QUERY_ID_NAME, session.getQueryId());
+
+        // Generate a storage table name
+        String storageTableName = "st" + UUID.randomUUID().toString().replace("-", "");
+        buildProps.put(STORAGE_TABLE, storageTableName);
+        buildProps.put(PRESTO_VIEW_FLAG, "true");
+        if (replace && definition.getComment().isPresent()) {
+            buildProps.put(TABLE_COMMENT, definition.getComment().get());
+        }
+        else {
+            // Add a default table comment when the view is being created
+            buildProps.put(TABLE_COMMENT, "Presto Materialized View");
+        }
+        properties = ImmutableMap.<String, String>builder()
+            .putAll(buildProps)
+            .build();
+
+        io.prestosql.plugin.hive.metastore.Column dummyColumn = new io.prestosql.plugin.hive.metastore.Column("dummy", HIVE_STRING, Optional.empty());
+
+        String schemaName = viewName.getSchemaName();
+        String tableName = viewName.getTableName();
+        Table.Builder tableBuilder = Table.builder()
+                .setDatabaseName(schemaName)
+                .setTableName(tableName)
+                .setOwner(session.getUser())
+                .setTableType(org.apache.hadoop.hive.metastore.TableType.VIRTUAL_VIEW.name())
+                .setDataColumns(ImmutableList.of(dummyColumn))
+                .setPartitionColumns(ImmutableList.of())
+                .setParameters(properties);
+        tableBuilder.getStorageBuilder()
+                .setStorageFormat(VIEW_STORAGE_FORMAT)
+                .setLocation("");
+
+        HiveIdentity identity = new HiveIdentity(session);
+        Optional<Table> existing = metastore.getTable(identity, viewName.getSchemaName(), viewName.getTableName());
+
+        // It's a create command where the materialized view already exists and 'if not exists' clause is specified
+        if (!replace && existing.isPresent() && !ignoreExisting) {
```

Comment:
This logic seems wrong. If `ignoreExisting` is true, we will still throw `MaterializedViewAlreadyExistsException` below. Should we return here instead? Something like
```java
if (!replace && existing.isPresent()) {
    if (ignoreExisting) {
        return;
    }
    throw new MaterializedViewAlreadyExistsException(viewName);
}
```

Reply:
How is `OR REPLACE` supposed to work in combination with `IF NOT EXISTS`? Why do we have both of these flags?

We don't seem to have any tests for `IF NOT EXISTS` (both with and without `OR REPLACE`)

Reply:
We talked offline. The expected semantics is: 
`The optional 'IF NOT EXISTS' clause causes the error to be suppressed if a materialized view already exists. This clause is ignored if 'OR REPLACE' is specified.`
We agreed that if both `or replace` and `if not exists` is specified, engine should throw an error. I will add a separate commit to this PR for that change. Will also add tests.